src/state/trajectory/logics.ts

```diff
@@ -334,21 +343,96 @@ const loadFileViaUrl = createLogic({
 });
 
 const fileConversionLogic = createLogic({
-    process(deps: ReduxLogicDeps, dispatch) {
+    process(deps: ReduxLogicDeps, dispatch, done) {
         dispatch(
             setStatus({
                 status: VIEWER_IMPORTING,
             })
         );
+        done();
     },
     type: CONVERT_FILE,
 });
 
+const setConversionEngineLogic = createLogic({
+    async process(deps: ReduxLogicDeps): Promise<{
+        engineType: AvailableEngines;
+        template: Template;
+        templateMap: TemplateMap;
+    }> {
+        const {
+            httpClient,
+            action,
+            uiTemplateUrlRoot,
+            uiBaseTypes,
+            uiCustomTypes,
+            uiTemplateDownloadUrlRoot,
+        } = deps;
+        const baseTypes = await httpClient
+            .get(`${uiTemplateDownloadUrlRoot}/${uiBaseTypes}`)
+            .then((baseTypesReturn: AxiosResponse) => {
+                console.log(baseTypesReturn);
+                return baseTypesReturn.data;
+            });
+
+        const customTypes = await httpClient
+            .get(`${uiTemplateUrlRoot}/${uiCustomTypes}`)
+            .then((customTypesReturn: AxiosResponse) => {
+                console.log(customTypesReturn);
+                return customTypesReturn.data;
+            })
+            .then((fileRefs) =>
+                Promise.all(
+                    map(
+                        fileRefs,
+                        async (ref) =>
+                            await httpClient
+                                .get(ref.download_url)
+                                .then((file) => file.data)
+                    )
+                )
+            );
+
+        const initTypeMap: TemplateMap = {};
+
+        const typeMap: TemplateMap = reduce(
+            customTypes,
+            (acc, cur: CustomTypeDownload) => {
+                //CustomType always has just one
+                const key = Object.keys(cur)[0] as string;
+                acc[key] = cur[key];
+                return acc;
+            },
+            initTypeMap
+        );
+        baseTypes["base_types"].forEach((type: BaseType) => {
+            typeMap[type.id] = { ...type, isBaseType: true };
+        });
+        const templateName =
+            ENGINE_TO_TEMPLATE_MAP[action.payload as AvailableEngines];
+        const engineTemplate = await httpClient
+            .get(`${uiTemplateDownloadUrlRoot}/${templateName}.json`)
+            .then((engineTemplateReturn) => engineTemplateReturn.data);
+        return {
+            engineType: action.payload,
+            template: engineTemplate[templateName],
+            templateMap: typeMap,
+        };
+    },
+    processOptions: {
+        dispatchReturn: true,
+        successType: SET_CONVERSION_TEMPLATE,
+        // failType: FETCH_POLLS_FAILED, // dispatch this failed action type
```

Comment:
can this line be removed or maybe add a comment about why it should stay if so

Reply:
not sure about this one. @meganrm?

Reply:
yeah. Oh yeah, I think we have a generic fail type you can set here